src/contexts/TodoContext.tsx

```diff
@@ -0,0 +1,231 @@
+import React, {
+  useState, useEffect, useCallback, useRef,
+} from 'react';
+import { TodoContextType } from '../types/TodoContextType';
+import { Todo } from '../types/Todo';
+import { TypeOfFiltering } from '../types/TypeOfFiltering';
+import { ErrorType } from '../types/ErrorType';
+import { client } from '../utils/fetchClient';
+import { UserWarning } from '../UserWarning';
+
+export const TodoContext = React.createContext<TodoContextType>({
+  todos: [],
+  tempTodo: null,
+  changeData: () => {},
+  deleteData: () => {},
+  filterType: TypeOfFiltering.All,
+  setFilterType: () => {},
+  dataError: '',
+  setError: () => {},
+  Error: () => {},
+  addTodo: () => {},
+  editTodo: -1,
+  setEditTodo: () => {},
+  changeTodo: () => {},
+  activeLoader: [],
+  shouldFocus: true,
+  setInputValue: () => {},
+  inputValue: '',
+  setTodos: () => {},
+});
+
+type Props = {
+  children: React.ReactNode,
+};
+
+const USER_ID = 9940;
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<TypeOfFiltering>(
+    TypeOfFiltering.All,
+  );
+  const [dataError, setError] = useState<string>('');
+  const [activeLoader, setActiveLoader] = useState<number[]>([]);
+  const [shouldFocus, setShouldFocus] = useState<boolean>(true);
+  const [editTodo, setEditTodo] = useState<number>(-1);
+  const [inputValue, setInputValue] = useState('');
+
+  const timeoutId = useRef<NodeJS.Timeout | null>(null);
+
+  const Error = useCallback((error: ErrorType) => {
+    setError(error);
+
+    timeoutId.current = setTimeout(() => {
+      setError('');
+    }, 3000);
+  }, []);
+
+  const getTodos = () => {
+    return client.get<Todo[]>(`/todos?userId=${USER_ID}`);
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch {
+        Error(ErrorType.Load);
+      }
+    };
+
+    fetchData();
+
+    return () => {
+      if (timeoutId.current !== null) {
+        clearTimeout(timeoutId.current);
+      }
+    };
+  }, [Error]);
+
+  if (!USER_ID) {
+    return <UserWarning />;
+  }
```

Comment:
it is very unexpected to return jsx inside `context`. you can create another wrapper and use it like this:
```
<ErrorCatcher>
  <App>
    ...
  </App>
</ErrorCatcher>
```
and do it inside ErrorCatcher